components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/RequestDispatcher.java

```diff
@@ -95,16 +105,13 @@ private void servePageOrFragment(App app, HttpRequest request, HttpResponse resp
                 JsonObject renderedFragment = app.renderFragment(request, response);
                 response.setContent(STATUS_OK, renderedFragment.toString(), CONTENT_TYPE_APPLICATION_JSON);
             } else {
-                // Request for a page.
-                // Logic to validate against CSRF attacks
-                if (request.getMethod().equals("POST") &&
-                        !app.getConfiguration().getCsrfIgnoreUris().contains(request.getUriWithoutContextPath())) {
-                    // POST request where the URI isn't in the CSRF ignore list, hence validate the CSRF Token
-                    if (request.getCookieValue(SessionRegistry.CSRF_TOKEN) == null ||
-                            request.getFormParams().get("uuf-csrftoken") == null ||
-                            !request.getFormParams().get("uuf-csrftoken").equals(
-                                    request.getCookieValue(SessionRegistry.CSRF_TOKEN))) {
-                        serveDefaultErrorPage(STATUS_UNAUTHORIZED, "CSRF threat detected", response);
+                // Execute filters
+                Configuration configuration = app.getConfiguration();
+                for (Filter filter : filters) {
+                    FilterResult result = filter.doFilter(configuration, request, response);
+                    if (!result.isContinue()) {
+                        serveDefaultErrorPage(result.getStatusCode().orElse(STATUS_INTERNAL_SERVER_ERROR),
```

Comment:
Instead of rendering the default error page, we should render configured error pages in the app `app.renderErrorPage`